src/events/events.service.ts

```diff
@@ -356,7 +356,7 @@ export class EventsService {
     if (url) {
       metadata = { ...metadata, url: url };
       const existingEvent = await this.getEventByUrl(url);
-      if (existingEvent) {
+      if (existingEvent && existingEvent.deleted_at == null) {
```

Comment:
Can this check be moved into `getEventByUrl`? For a given URL, if we have an event with a null `deleted_at` and one with a non-null `deleted_at`, the query can currently return the deleted event.